src/sagemaker/tensorflow/model.py

```diff
@@ -396,8 +396,8 @@ def prepare_container_def(
             # model is not yet there, defer repacking to later during pipeline execution
             if isinstance(self.sagemaker_session, PipelineSession):
                 self.sagemaker_session.context.need_runtime_repack.add(id(self))
-                self.sagemaker_session.context.runtime_repack_output_prefix = "s3://{}/{}".format(
-                    bucket, key_prefix
+                self.sagemaker_session.context.runtime_repack_output_prefix = s3.s3_path_join(
+                    "s3://", bucket, key_prefix
                 )
```

Comment:
Seems the `s3.s3_path_join` will remove the "/" in the end of the path and return "s3://my-bucket/my-key-prefix".

But given the customer input in https://github.com/aws/sagemaker-python-sdk/issues/3413
```
Removing the slash from the code location “works” (as in does not create an error message), but then I have no separation between the pipeline prefix and the artifact name, e.g. pipeline-codepytorchinference ...
```
Should we also check the estimator side on how it generates the entire code output path with the `code_location`?

Or if this is already checked in the unit test that a separation is correctly added between the prefix and the artifact name, please let me know. Then we're good

Reply:
Plus, if we decide to update the model side rather than the estimator side, as this only updates the tensorflow model, same changes are needed in the model class as well:
- src/sagemaker/model.py: https://github.com/aws/sagemaker-python-sdk/pull/3257/files#diff-25e5d90876e4e075cba7fe8ae004e6c43d300faeecc2d076d1babacc5c0f6d06

Reply:
>Or if this is already checked in the unit test that a separation is correctly added between the prefix and the artifact name, please let me know. Then we're good
yep, I verified in the unit test that the s3 path will be `s3://bucket/pipeline-code/pytorchinference/...`. Basically this code change will convert the `//` to `/`
>same changes are needed in the model class as well
Yes, forgot to commit that class adding it now

Reply:
After a second thought, how about we directly update the joins in `_stage_user_code_in_s3` in estimator.py (see [here](https://github.com/aws/sagemaker-python-sdk/blob/322b919c3b7eaee94adf2a1d9df6e25e756ce55b/src/sagemaker/estimator.py#L808))? The reasons are:
- even if we update the model side to remove the "/" at the end, the `code_location//some-key` would still persist in estimator code output. Though it may not cause any errors but it may not work as customer's expectation.
- If update the estimator side, we can just update in one place and no need to update the model sides.

Reply:
I am assuming that by "artifact" the user means "model-name"

Reply:
> how about we directly update the joins in _stage_user_code_in_s3 in estimator.py

Synced up offline, this may break backward compatibility. Given that we don't see any customer complains on the estimator's code location output path, we can just simply update the model side